tests/test_backend_pyomo.py

```diff
@@ -2365,6 +2366,51 @@ def test_update_parameter_replace_undefined(self, caplog, simple_supply):
         default_val = simple_supply._model_data.attrs["defaults"]["source_eff"]
         assert expected.equals(updated_param.fillna(default_val))
 
+    def test_update_parameter_refs_in_order(self, caplog, simple_supply):
+        """
+        `cost_flow_cap` for test_demand_elec is not defined initially.
+        We update `cost_flow_cap` to zero for all index values
+        and then check that it has propagated through to `cost_investment_flow_cap`, `cost`, and the objective function.
+        """
+
+        caplog.set_level(logging.DEBUG)
+        simple_supply.backend.verbose_strings()
+
+        def _check_components(true_or_false: bool):
```

Comment:
A tad verbose... I'd say the stages before `update_parameter` are not needed (or a separate test?).

Suggestions:
- I'd keep the docstring to a short sentence, like "Check that parameter re-definition propagates to constraints and obj. function"
- re-arrange so that asserts are done once, not inside `for` loops.
- avoid defining functions inside the test.
- Change 0 (which could be a default) to something rare. I like `0xDEADBEEF`, a common debug integer.